tests/testOutputRoot.py

```diff
@@ -65,27 +62,37 @@ class OutputRootTestCase(unittest.TestCase):
     """A test case for output roots."""
 
     def setUp(self):
-        # Note: these will succeed even if the directories don't exist.
-        subprocess.call(["rm", "-rf", testOutput])
-        subprocess.call(["rm", "-rf", testOutput2])
-        subprocess.call(["rm", "-rf", testInput1])
-        subprocess.call(["rm", "-rf", testInput2])
+        self.tempDirs = {}
 
     def tearDown(self):
-        subprocess.call(["rm", "-rf", testOutput, testOutput2,
-                         testOutput3, testInput1, testInput2])
+        for d in self.tempDirs:
+            shutil.rmtree(self.tempDirs[d])
+
+    def mkdtemp(self, prefix):
+        """Create a temporary directory and return its name.
+        The resulting path is stored in a dict (self.tempDirs) with
+        the supplied prefix as key. This allows the name to be retrieved
+        later. The directory path is returned.
+        A '-' is appended to the prefix if not there."""
+        dprefix = prefix
+        if not dprefix.endswith('-'):
+            dprefix = dprefix + '-'
+        tempDir = tempfile.mkdtemp(dir=ROOT, prefix=dprefix)
+        self.tempDirs[prefix] = tempDir
+        return tempDir
```

Comment:
It seems clearer to me so I would like to keep it. I didn't want to create a bunch of temp directories that were not being used so I switched to a test only creating a temp directory when it really needed one. I think the real solution would be to add something to `lsst.utils.tests` to create a temp directory in a context manager.